plutus-benchmark/script-contexts/src/PlutusBenchmark/ScriptContexts.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE BangPatterns      #-}
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+
+module PlutusBenchmark.ScriptContexts where
+
+import PlutusLedgerApi.V1
+import PlutusLedgerApi.V1.Address
+import PlutusLedgerApi.V1.Value
+import PlutusTx qualified as PlutusTx
+import PlutusTx.Builtins qualified as PlutusTx
+import PlutusTx.Prelude qualified as PlutusTx
+
+-- | A very crude deterministic generator for 'ScriptContext's with size
+-- approximately proportional to the input integer.
+mkScriptContext :: Int -> ScriptContext
+mkScriptContext i = ScriptContext (mkTxInfo i) (Spending (TxOutRef (TxId "") 0))
+
+mkTxInfo :: Int -> TxInfo
+mkTxInfo i = TxInfo {
+  txInfoInputs=mempty,
+  txInfoOutputs=fmap mkTxOut [1..i],
+  txInfoFee=mempty,
+  txInfoMint=mempty,
+  txInfoDCert=mempty,
+  txInfoWdrl=mempty,
+  txInfoValidRange=always,
+  txInfoSignatories=mempty,
+  txInfoData=mempty,
+  txInfoId=TxId ""
+  }
+
+mkTxOut :: Int -> TxOut
+mkTxOut i = TxOut {
+  txOutAddress=pubKeyHashAddress (PubKeyHash ""),
+  txOutValue=mkValue i,
+  txOutDatumHash=Nothing
+  }
+
+mkValue :: Int -> Value
+mkValue i = assetClassValue (assetClass adaSymbol adaToken) (fromIntegral i)
+
+-- This example decodes the script context (which is O(size-of-context) work), and then also
+-- does some work that's roughly proportional to the size of the script context (counting the
+-- outputs). This should be a somewhat realistic example where a reasonable chunk of work is
+-- done in addition to decoding.
+{-# INLINABLE checkScriptContext1 #-}
+checkScriptContext1 :: PlutusTx.BuiltinData -> ()
```

Comment:
The names could be more meaningful, but I'm just saying that for the sake of saying something.